packages/shared/src/components/squads/SquadListingHeader.spec.tsx

```diff
@@ -0,0 +1,84 @@
+import { render, RenderResult, screen } from '@testing-library/preact';
+import { QueryClient, QueryClientProvider } from 'react-query';
+import React from 'react';
+import nock from 'nock';
+import { IFlags } from 'flagsmith';
+import { AuthContextProvider } from '../../contexts/AuthContext';
+import loggedUser from '../../../__tests__/fixture/loggedUser';
+import { generateTestSquad } from '../../../__tests__/fixture/squads';
+import { FeaturesContextProvider } from '../../contexts/FeaturesContext';
+import { SquadListingHeader } from './SquadListingHeader';
+
+let features: IFlags;
+Object.defineProperty(window, 'matchMedia', {
+  writable: true,
+  value: jest.fn().mockImplementation((query) => ({
+    matches: true,
+    media: query,
+    onchange: null,
+    addListener: jest.fn(), // Deprecated
+    removeListener: jest.fn(), // Deprecated
+    addEventListener: jest.fn(),
+    removeEventListener: jest.fn(),
+    dispatchEvent: jest.fn(),
+  })),
+});
+
+const defaultFeatures: IFlags = {
+  squad: {
+    enabled: true,
+  },
+};
+
+beforeEach(async () => {
+  nock.cleanAll();
+  jest.clearAllMocks();
+  features = defaultFeatures;
+});
+
+const squads = [generateTestSquad()];
+
+const renderComponent = (isOwner = false): RenderResult => {
+  const client = new QueryClient();
+
+  return render(
+    <QueryClientProvider client={client}>
+      <FeaturesContextProvider flags={features}>
+        <AuthContextProvider
+          user={loggedUser}
+          updateUser={jest.fn()}
+          tokenRefreshed
+          getRedirectUri={jest.fn()}
+          loadingUser={false}
+          loadedUserFromCache
+          squads={squads}
+        >
+          <SquadListingHeader isOwner={isOwner} />
+        </AuthContextProvider>
+      </FeaturesContextProvider>
+    </QueryClientProvider>,
+  );
+};
+
+// Renders the component with create squad button
+it('should render the component as a squad user', async () => {
+  renderComponent();
+  const alertMock = jest.spyOn(window, 'alert').mockImplementation(() => {});
```

Comment:
Why do we test on alerts, it doesn't seem like a good test case or implementation actually.